app/services/the_tvdb_service/row.rb

```diff
@@ -0,0 +1,52 @@
+class TheTvdbService
+  class Row
+    attr_reader :media_id, :media_type, :data, :episode_id, :tvdb_series_id
+
+    def initialize(media_id, data, tvdb_series_id)
+      @media_id = media_id
+      @data = data
+      @episode_id = data['id']
+      @tvdb_series_id = tvdb_series_id
+      @media_type = 'Anime'
+    end
+
+    def update_episode!
+      return unless episode_number.present?
+
+      episode.season_number ||= data['airedSeason']
+      episode.relative_number ||= data['airedEpisodeNumber']
+      # was returning an empty string and storing that in database.
+      episode.synopsis = data['overview'].presence || nil if episode.synopsis.blank?
+      episode.thumbnail_file_name ||= thumbnail_path
+      episode.airdate ||= data['firstAired']
+      # HACK: due to some corrupt db entries having {"en_jp" => nil} which causes validation errors
+      episode.titles['en_jp'] ||= data['episodeName'] || "Episode #{episode.number}"
```

Comment:
Most of these would do well to be put in brief methods of their own, such as this:

```ruby
def episode_name
  data['episodeName'] || "Episode #{episode.number}"
end
```